grafana/grafana
# Elasticsearch: Fix top metrics with terms in backend

Part of https://github.com/grafana/grafana/issues/54011. During working on https://github.com/grafana/grafana/pull/64540/.This PR fixes processing of percentiles on backend I noticed that in https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/elasticsearch/ElasticResponse.ts#L231-L244 we have processing for `top_metrics`, which is missing on backend. This PR adds it (the code is pretty much the same) and adds test. 

How to test: 

1. Run elastic data source `make devenv sources=elastic`
2. Open Explore 
3. With `shouldRunTroughBackend` in `elastc/datasource.ts` set to `false`, run [top_metrics + terms query](http://localhost:3000/explore?orgId=1&left=%7B%22datasource%22:%22PE50363A9B6833EE7%22,%22queries%22:%5B%7B%22alias%22:%22%22,%22bucketAggs%22:%5B%7B%22field%22:%22level%22,%22id%22:%222%22,%22settings%22:%7B%22min_doc_count%22:%221%22,%22order%22:%22desc%22,%22orderBy%22:%22_term%22,%22size%22:%2210%22%7D,%22type%22:%22terms%22%7D%5D,%22datasource%22:%7B%22type%22:%22elasticsearch%22,%22uid%22:%22PE50363A9B6833EE7%22%7D,%22metrics%22:%5B%7B%22id%22:%221%22,%22settings%22:%7B%22metrics%22:%5B%22counter%22,%22float%22%5D,%22order%22:%22desc%22,%22orderBy%22:%22counter%22%7D,%22type%22:%22top_metrics%22%7D%5D,%22query%22:%22%22,%22refId%22:%22A%22,%22timeField%22:%22@timestamp%22%7D%5D,%22range%22:%7B%22from%22:%22now-6h%22,%22to%22:%22now%22%7D%7D) 

<img width="1381" alt="image" src="https://user-images.githubusercontent.com/30407135/226404591-04b461c9-8cf9-4975-9d9e-b0c19d488812.png">
4. Set `shouldRunTroughBackend` in `elastc/datasource.ts` to `true`, run same query. The results should be the same.
